### PR TITLE
Update geth version

### PR DESCRIPTION
**Pull request by simonlechner**
This PR switches from the heavily modified `go-ethereum-substate` to the new `tosca-go-ethereum' as the geth source. 
With this new geth version it can be used to introduce new revision support to Tosca.

### CODE REVIEW

**Review by facuMH — COMMENTED** [review]
a couple of minor comments.

File: go/vm/geth/geth.go
Comment (simonlechner) [reply go/vm/geth/geth.go:258]: Very good suggestion, added a panic to all unimplemented functions.

File: go/geth_adapter/adapter.go
Comment (HerbertJordan) [inline go/geth_adapter/adapter.go:499]: This might be future candidates for public utility functions in the `vm` package.

File: go/geth_adapter/adapter.go
Comment (HerbertJordan) [inline go/geth_adapter/adapter.go:491]: This should be impossible considering the input being a 256 bit value. Considering this, and to return 0 for a nil value, would allow to eliminate the error return parameter. This would enable the function to be used in expressions.

File: go/vm/geth/geth.go
Comment (HerbertJordan) [inline go/vm/geth/geth.go:145]: This could benefit from offering a `ToUint256()` member function for the `Value` type.

File: go/vm/geth/geth.go
Comment (HerbertJordan) [inline go/vm/geth/geth.go:203]: Also this line here.

File: go/vm/lfvm/converter.go
Comment (HerbertJordan) [inline go/vm/lfvm/converter.go:559]: `PREFRANDAO` should probably not be in this list since it is actually covered (just named differently)

**Review by HerbertJordan — COMMENTED** [review]
Looks good, but we should wait for merging it until the tosca-go-ethereum PR is merged.

File: go/geth_adapter/adapter.go
Comment (simonlechner) [reply go/geth_adapter/adapter.go:499]: I think this should be done in a separate PR to not move unrelated code in this PR.

File: go/vm/geth/geth.go
Comment (simonlechner) [reply go/vm/geth/geth.go]: I moved the `Uint256ToValue` and `ValueToUint256` functions to the `vm/utils.go` file and reused them here.

**Comment by simonlechner** [comment]
A full CT run passed with the current version of this PR and the `tosca-go-ethereum` repository. 
```
[t=1872:40] - Processing ~484k tests per second, total 58180515212, skipped 0, found issues 0
[t=1872:45] - Processing ~295k tests per second, total 58181991102, skipped 0, found issues 0
[t=1872:50] - Processing ~338k tests per second, total 58183679524, skipped 0, found issues 0
[t=1872:55] - Processing ~343k tests per second, total 58185396597, skipped 0, found issues 0
All tests passed successfully!
```

File: go/vm/geth/ct.go
Comment (HerbertJordan) [inline go/vm/geth/ct.go:168]: Is this a use case for `vm.Uint256ToValue`?

Also in the other functions below?

File: go/vm/utils.go
Comment (HerbertJordan) [inline go/vm/utils.go:72]: As it is part of the `vm` package, these functions are part of the public interface of Tosca. Thus, they
- should be documented
- should have unit tests
Even if seemingly simple, byte-order issues may easily happen. Also, tests document their intended use.

**Review by HerbertJordan — CHANGES_REQUESTED** [review]

File: go/vm/geth/geth.go
Comment (facuMH) [inline go/vm/geth/geth.go:186]: panic here as well ?

File: go/vm/geth/geth.go
Comment (facuMH) [inline go/vm/geth/geth.go:253]: panic ?

**Review by facuMH — CHANGES_REQUESTED** [review]

File: go/vm/geth/geth.go
Comment (simonlechner) [reply go/vm/geth/geth.go:186]: All the unimplemented functions without a `panic` are called during at least one test and should not terminate the program execution.

File: go/vm/geth/ct.go
Comment (HerbertJordan) [inline go/vm/geth/ct.go:168]: The Clone should be unnecessary, right?
```suggestion
		Value:     vm.Uint256ToValue(value),
```
Also in functions below.

File: go/vm/utils_test.go
Comment (HerbertJordan) [inline go/vm/utils_test.go:30]: nit: should be the same as
```suggestion
	expected := [32]byte{}
```

**Review by HerbertJordan — APPROVED** [review]
Looks good, thanks!

File: go/vm/utils_test.go
Comment (facuMH) [inline go/vm/utils_test.go:18]: nit: add an empty line after a function.

**Review by facuMH — APPROVED** [review]
nice to get closer to the update!!

**Review by facuMH — APPROVED** [review]

**Review by HerbertJordan — APPROVED** [review]

**Pull request merged**
